app/javascript/src/components/Dashboard/Contacts/index.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState, useEffect } from "react";
+
+import { Button, PageLoader } from "neetoui";
+import { Container, Header } from "neetoui/layouts";
+
+import contactsApi from "apis/contacts";
+
+import DeleteAlert from "./DeleteAlert";
+import MenuBar from "./Menubar";
+import NewContactPane from "./Pane/Create";
+import Table from "./Table";
+
+const Contact = () => {
+  const [loading, setLoading] = useState(true);
+  const [showNewContactPane, setShowNewContactPane] = useState(false);
+  const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [selectedContactId, setSelectedContactId] = useState(null);
+  const [contacts, setContacts] = useState([]);
+  const [showMenu, setShowMenu] = useState(true);
+
+  useEffect(() => {
+    fetchContacts();
+  }, []);
+
+  const fetchContacts = async () => {
+    try {
+      setLoading(true);
+      const { data } = await contactsApi.fetch();
+      setContacts(data);
+    } catch (error) {
+      logger.error(error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleDelete = contactId => {
+    setSelectedContactId(contactId);
+    setShowDeleteAlert(true);
+  };
+
+  const onMenuBarToggle = () => {
+    setShowMenu(prevShowMenu => !prevShowMenu);
+  };
```

Comment:
Do we need this to be separate function? Are we using this at more than one place? If not, I think we should just invoke this setter wherever required. The setter name is pretty self-explanatory.